lib/elixir/test/elixir/map_test.exs

```diff
@@ -388,6 +388,18 @@ defmodule MapTest do
     end
   end
 
+  test "defstruct warns with duplicate keys" do
+    import ExUnit.CaptureIO
+
+    assert capture_io(:stderr, fn ->
```

Comment:
If `:stderr` is used, tests cannot be `async`. I would move this test to [the `warnings_test.exs` file](https://github.com/elixir-lang/elixir/blob/master/lib/elixir/test/elixir/kernel/warning_test.exs)

Reply:
I did wonder why there was a separate file for warnings, that explains it!